gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteUniqueNamesValidatorTest.java

```diff
@@ -0,0 +1,206 @@
+  private final GtfsRoute routeOnlyLongName_alpha =
+      new GtfsRoute.Builder()
+          .setCsvRowNumber(2)
+          .setRouteId("routeOnlyLongName_alpha")
+          .setRouteLongName("alpha")
+          .build();
+  private final GtfsRoute routeOnlyShortName_abcd =
+      new GtfsRoute.Builder()
+          .setCsvRowNumber(3)
+          .setRouteId("routeOnlyShortName_abcd")
+          .setRouteShortName("abcd")
+          .build();
+  private final GtfsRoute routeLongName_ab_ShortName_cd =
+      new GtfsRoute.Builder()
+          .setCsvRowNumber(4)
+          .setRouteId("routeLongName_ab_ShortName_cd")
+          .setRouteLongName("ab")
+          .setRouteShortName("cd")
+          .build();
+  private final GtfsRoute routeLongName_cd_ShortName_ab =
+      new GtfsRoute.Builder()
+          .setCsvRowNumber(5)
+          .setRouteId("routeLongName_cd_ShortName_ab")
+          .setRouteLongName("cd")
+          .setRouteShortName("ab")
+          .build();
+  private final GtfsRoute routeLongName_abc_ShortName_d =
+      new GtfsRoute.Builder()
+          .setCsvRowNumber(6)
+          .setRouteId("routeLongName_abc_ShortName_d")
+          .setRouteLongName("abc")
+          .setRouteShortName("d")
+          .build();
+
+  @Test
+  public void routeOnlyLongNameEqualsAnotherRouteOnlyShortName() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteUniqueNamesValidator validator = new RouteUniqueNamesValidator();
+
+    List<GtfsRoute> routes = new ArrayList<>();
+    routes.add(routeOnlyLongName_abcd);
+    routes.add(routeOnlyShortName_abcd);
+    validator.routeTable = GtfsRouteTableContainer.forEntities(routes, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices())
+        .containsExactly(
+            new DuplicateRouteLongNameShortNameCombinationNotice(
+                "routeOnlyLongName_abcd", "abcd", "", "routeOnlyShortName_abcd", "", "abcd"));
+  }
+
+  @Test
+  public void routeOnlyLongNameDifferentFromAnotherRouteOnlyShortName() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteUniqueNamesValidator validator = new RouteUniqueNamesValidator();
+
+    List<GtfsRoute> routes = new ArrayList<>();
+    routes.add(routeOnlyLongName_alpha);
+    routes.add(routeOnlyShortName_abcd);
+    validator.routeTable = GtfsRouteTableContainer.forEntities(routes, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices()).isEmpty();
+  }
+
+  @Test
+  public void routeOnlyLongNameDifferentFromAnotherRouteOnlyLongName() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteUniqueNamesValidator validator = new RouteUniqueNamesValidator();
+
+    List<GtfsRoute> routes = new ArrayList<>();
+    routes.add(routeOnlyLongName_abcd);
+    routes.add(routeOnlyLongName_alpha);
+    validator.routeTable = GtfsRouteTableContainer.forEntities(routes, noticeContainer);
```

Comment:
Can you use Arrays.asList to make the code more readable?

Reply:
Sure. Good idea! Has been fixed!